msg_queue.go

```diff
@@ -34,6 +95,36 @@ func (m *msgQueue) readMessage(st State, current *View) *MessageReq {
 	return msg
 }
 
+func readValues(queue *msgQueueImpl, current *View) (*MessageReq, []*MessageReq) {
+	discarded := []*MessageReq{}
```

Comment:
If we are going to keep this function, keep in mind that we need lock here as well, because `PushMessage` is invoked from a separate routine.